pkgs/crane-utils/src/bin/crane-resolve-workspace-inheritance.rs

```diff
@@ -105,90 +96,300 @@ fn merge(cargo_toml: &mut toml::Value, root: &toml::Value) {
         ("dev-dependencies", "dependencies"),
         ("build-dependencies", "dependencies"),
     ] {
-        if let (Some(toml::Value::Table(p)), Some(toml::Value::Table(wp))) =
-            (t.get_mut(key), w.get(ws_key))
-        {
-            merge_tables(p, wp);
+        if let Some((cargo_toml, root)) = cargo_toml.get_mut(key).zip(w.get(ws_key)) {
+            try_merge_cargo_tables(cargo_toml, root);
         };
 
-        if let Some(toml::Value::Table(targets)) = t.get_mut("target") {
-            for (_, tp) in targets {
-                if let (Some(toml::Value::Table(p)), Some(toml::Value::Table(wp))) =
-                    (tp.get_mut(key), w.get(ws_key))
-                {
-                    merge_tables(p, wp);
-                };
+        if let Some(targets) = cargo_toml.get_mut("target").and_then(try_as_table_like_mut) {
+            for (_, tp) in targets.iter_mut() {
+                if let Some((cargo_toml, root)) = tp.get_mut(key).zip(w.get(ws_key)) {
+                    try_merge_cargo_tables(cargo_toml, root);
+                }
             }
         }
     }
+
+    // cleanup stray spaces left by merging InlineTable values into Tables
+    clear_value_formatting(cargo_toml.as_item_mut());
+}
+
+/// Return a [`toml_edit::TableLike`] representation of the [`Item`] (if any)
+fn try_as_table_like(item: &Item) -> Option<&dyn toml_edit::TableLike> {
+    match item {
+        Item::Table(w) => Some(w),
+        Item::Value(toml_edit::Value::InlineTable(w)) => Some(w),
+        _ => None,
+    }
 }
 
-fn merge_tables(cargo_toml: &mut Table, root: &Table) {
+/// Return a mutable [`toml_edit::TableLike`] representation of the [`Item`] (if any)
+fn try_as_table_like_mut(item: &mut Item) -> Option<&mut dyn toml_edit::TableLike> {
+    match item {
+        Item::Table(w) => Some(w),
+        Item::Value(toml_edit::Value::InlineTable(w)) => Some(w),
+        _ => None,
+    }
+}
+
+/// Merge the specified `cargo_toml` and workspace `root` if both are tables
+fn try_merge_cargo_tables(cargo_toml: &mut Item, root: &Item) {
+    match (cargo_toml, root) {
+        (
+            Item::Table(p), //
```

Comment:
Missing comment? Or is this to nudge rustfmt?

Reply:
This was a nudge to rustfmt, but removed in simplifying to use dyn instead, per the other comment.